src/EnergyPlus/WaterUse.cc

```diff
@@ -1496,42 +1496,45 @@ namespace WaterUse {
 
         } else { // WaterConnections(WaterConnNum)%TotalMassFlowRate > 0.0
 
-            {
-                auto const SELECT_CASE_var(this->HeatRecoveryConfig);
-                if (SELECT_CASE_var == HeatRecoveryConfigEnum::Plant) {
-                    this->RecoveryMassFlowRate = this->HotMassFlowRate;
-                } else if (SELECT_CASE_var == HeatRecoveryConfigEnum::Equipment) {
-                    this->RecoveryMassFlowRate = this->ColdMassFlowRate;
-                } else if (SELECT_CASE_var == HeatRecoveryConfigEnum::PlantAndEquip) {
-                    this->RecoveryMassFlowRate = this->TotalMassFlowRate;
-                }
+            switch (this->HeatRecoveryConfig) {
```

Comment:
I don't know what else to say as I review this, it's just so much better.  Every SELECT_CASE_var removed brings one extra smile to someone, somewhere.